pkg/glusterblock/controllerserver.go

```diff
@@ -0,0 +1,316 @@
+	}
+
+	volumeName := parseResp.gdVolReq.Name
+
+	blockVolInfo := bapi.BlockVolumeInfo{
+		Name: volumeName,
+		Size: uint64(volSizeBytes),
+	}
+	volumeReq := bapi.BlockVolumeCreateRequest{
+		BlockVolumeInfo: &blockVolInfo,
+	}
+
+	glog.V(2).Infof("block volume create request: %+v", volumeReq)
+	blockVolCreateResp, err := cs.client.BlockVolumeCreate("gluster-loopback", volumeReq)
+	if err != nil {
+		glog.Errorf("failed to create block volume: %s err: %v", volumeName, err)
+		errResp := cs.client.LastErrorResponse()
+		// errResp will be nil in case of `No route to host` error
+		if errResp != nil && errResp.StatusCode == http.StatusConflict {
+			return nil, status.Errorf(codes.Aborted, "block volume create already in process: %v", err)
+		}
+
+		return nil, status.Errorf(codes.Internal, "failed to create block volume: %s err: %v", volumeName, err)
+	}
+
+	glusterServer, bkpServers, err := utils.GetClusterNodes(cs.client)
+	if err != nil {
+		glog.Errorf("failed to get cluster nodes: %v", err)
+		return nil, status.Errorf(codes.Internal, "failed to get cluster nodes: %v", err)
+	}
+
+	glog.V(3).Infof("blockvolume create response : %+v", blockVolCreateResp)
+	resp := &csi.CreateVolumeResponse{
+		Volume: &csi.Volume{
+			VolumeId:      volumeName,
+			CapacityBytes: volSizeBytes,
+			VolumeContext: map[string]string{
+				"glustervol":        blockVolCreateResp.HostingVolume,
+				"glusterserver":     glusterServer,
+				"glusterbkpservers": strings.Join(bkpServers, ":"),
+			},
+		},
+	}
+
+	glog.V(4).Infof("CSI block volume response: %+v", resp)
+	return resp, nil
+}
+
+func (cs *ControllerServer) getVolumeSize(req *csi.CreateVolumeRequest) int64 {
+	// If capacity mentioned, pick that or use default size 1 GB
+	volSizeBytes := defaultVolumeSize
+	if capRange := req.GetCapacityRange(); capRange != nil {
+		volSizeBytes = capRange.GetRequiredBytes()
+	}
+	return volSizeBytes
+}
+
+func (cs *ControllerServer) validateCreateVolumeReq(req *csi.CreateVolumeRequest) error {
+	if req == nil {
+		return status.Errorf(codes.InvalidArgument, "request cannot be empty")
+	}
+
+	if req.GetName() == "" {
+		return status.Error(codes.InvalidArgument, "name is a required field")
+	}
+
+	if reqCaps := req.GetVolumeCapabilities(); reqCaps == nil {
+		return status.Error(codes.InvalidArgument, "volume capabilities is a required field")
+	}
+
+	return nil
+}
+
+// DeleteVolume deletes the given volume.
+func (cs *ControllerServer) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequest) (*csi.DeleteVolumeResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "block volume delete request is nil")
+	}
+
+	volumeID := req.GetVolumeId()
+	if volumeID == "" {
+		return nil, status.Error(codes.InvalidArgument, "volume ID is nil")
+	}
+	glog.V(2).Infof("deleting block volume with ID: %s", volumeID)
+
+	// Delete volume
+	err := cs.client.BlockVolumeDelete("gluster-loopback", volumeID)
+	if err != nil {
```

Comment:
can be written as 

```
if err != nil {
		errResp := cs.client.LastErrorResponse()
		// errResp will be nil in case of No route to host error
		if errResp != nil && errResp.StatusCode != http.StatusNotFound {
			glog.Errorf("deleting block volume %s failed: %v", req.VolumeId, err)
		return nil, status.Errorf(codes.Internal, "deleting volume %s failed: %v", req.VolumeId, err)
		}
		
	}
        glog.Infof("successfully deleted block volume %s", volumeID)
	return &csi.DeleteVolumeResponse{}, nil
```

Reply:
This will print the successful block deletion message in case of failure as well. Hence retaining the existing check.